### PR TITLE
IGNITE-20337 Ensure QueryChecker is properly used in tests

### PR DESCRIPTION
**Pull request by AMashenkov**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Comment by lowka** [comment]
objects:
- creates instances (via some annotated method that creates instances of QueryChecker) (1)
- checks that `check` was call for those instances in after each callbacks
(1)
>
> `@CreateChecker("makeChecker")`
> public class MyTest ... {
> ..
> public QueryChecker makeChecker(String query, Object... params) {
>   ...
>   return new QueryChecker(...)
> }

**Review by zstan — APPROVED** [review]

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/QueryCheckerExtension.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ * JUnit extension to inject {@link QueryCheckerFactory} instance into test classes, and ensure the {@link QueryChecker#check()} method is
+ * called for each {@link QueryChecker} instance, which was created via the factory.
+ *
+ * @see InjectQueryCheckerFactory
+ */
+public class QueryCheckerExtension implements BeforeEachCallback, BeforeAllCallback, AfterEachCallback {
+    /** QueryCheckers instances that are managed by this extension. */
+    private static final Set<QueryChecker> QUERY_CHECKERS = new HashSet<>();
+
+    private static final QueryCheckerFactory FACTORY_INSTANCE = new QueryCheckerFactoryImpl(
```
Comment (lowka) [inline modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/QueryCheckerExtension.java:46]: @AMashenkov I think we can safely convert all static fields to instances fields w/o losing functionality.

Comment (lowka) [reply modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/QueryCheckerExtension.java:46]: I am talking about fields that store state. That applies to methods private static methods as well.

Comment (AMashenkov) [reply modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/QueryCheckerExtension.java]: Fixed

**Review by lowka — APPROVED** [review]

**Review by zstan — APPROVED** [review]

**Pull request merged**